File: algorithm/problem/linklist/83. 删除排序链表中的重复元素.cpp
// 给定一个已排序的链表的头 head ， 删除所有重复的元素，使每个元素只出现一次 。返回 已排序的链表 。

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (!head)
        {
            return head;
        }
        ListNode* currentNode = head;
        while (currentNode && currentNode->next)
        {
            if (currentNode->val == currentNode->next->val)
            {
                currentNode->next = currentNode->next->next;
            } else {
                currentNode = currentNode->next;
            }
        }
        return head;
    }
};

File: algorithm/problem/linklist/83. 删除排序链表中的重复元素.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
	Prev *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	var currentNode *ListNode = head
	for currentNode != nil && currentNode.Next != nil {
		if currentNode.Val == currentNode.Next.Val {
			currentNode.Next = currentNode.Next.Next
		} else {
			currentNode = currentNode.Next
		}
	}
	return head
}
